packages/imperative/src/plugins/AbstractPluginLifeCycle.ts

```diff
@@ -0,0 +1,51 @@
+/*
+* This program and the accompanying materials are made available under the terms of the
+* Eclipse Public License v2.0 which accompanies this distribution, and is available at
+* https://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Copyright Contributors to the Zowe Project.
+*
+*/
+
+/**
+ * An optional module of a plugin extends this class to
+ * perform actions during the installation lifecycle of a plugin.
+ * A plugin identifies that it supplies a life cycle class
+ * by specifying the 'pluginLifeCycle' imperative configuration property.
+ */
+export abstract class AbstractPluginLifeCycle {
+    /**
+     * This function will be called after a plugin is installed.
+     * A plugin can use this opportunity to perform a sanity test or to
+     * perform some additional setup which is specific to that plugin.
+     * A plugin which provides a specialized credential manager should use
+     * this opportunity to insert itself as an override of the standard
+     * credential manager that is delivered with Zowe CLI.
+     *
+     * @throws
+     *      An ImperativeError containing a message describing any error
+     *      that occurred while performing post-install actions.
+     */
+    public abstract postInstall(): void;
```

Comment:
Should we define the return type as `Promise<void> | void`, and await this method when we call it? Same question applies to the `preUninstall` hook.

I think we should support both synchronous and asynchronous code in these hooks. So if a plugin wants to run asynchronous code in their lifecycle hook, we'd want to wait for the promises to resolve.

Reply:
This seems really valuable. I made the suggested changes.